### PR TITLE
chore(eng): improve rolling

### PR DESCRIPTION
**Pull request by avodovnik**
This PR introduces the submodule for upstream dependency. The intent here is to help us ensure that we have a "stable" state of the docs from which our API was generated from, etc.

Additionally, it introduces a build task helper that encapsulates a bunch of common commands (like downloading the driver, generating the API, etc.) which are common during the dev process.

### CODE REVIEW

**Comment by mxschmitt** [comment]
I'm worried about the submodule, because in theory we already have a relation between the driver (commitinfo file) and the upstream repository. I would maybe add a bot like this to ensure the generated code is up2date which should fit for your needs:

```yml
  verify_generated_files:
    name: Verify Generated Files
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v2
      - name: Setup .NET Core
        uses: actions/setup-dotnet@v1
        with:
          dotnet-version: 5.0.100
      - name: Install dotnet-format
        run: dotnet tool install --global --version 5.0.211103 dotnet-format
      - name: Download Drivers
        run: dotnet run -p ./src/tools/Playwright.Tooling/Playwright.Tooling.csproj -- download-drivers --basepath .
      - name: Read commit SHA
        id: read-upstream-commit-sha
        run: echo "::set-output name=hash::$(cat src/Playwright/.drivers/linux/package/commitinfo)"
      - uses: actions/checkout@v2
        with:
          repository: 'microsoft/playwright'
          ref: ${{ steps.read-upstream-commit-sha.outputs.hash }}
          path: 'playwright'
      - name: Generate
        run: |
          node playwright/utils/doclint/generateDotnetApi.js src/Playwright
          rm -rf playwright
      - name: Verify generated API is up to date
        run: git diff --exit-code
```

Then instead of maintaining an upstream revision number (git submodule) and the driver version you only need to take care about a driver version like its done in Java and Python and they are always in sync since the bot needs to be green.

**Comment by avodovnik** [comment]
I've added the integrity check into #1620 but it uses the submodule.

File: build.ps1 [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+function Get-SubmoduleStatus() {
+  $smStatus = git submodule status
+  return $smStatus
+}
+
+function Invoke-InitializeSubmodule($enableReset = $true) {
+  # Check the status of the submodule, if not initialized, let's do that
+  if ((Get-SubmoduleStatus).StartsWith("-") -or ($enableReset -and $reset)) {
+    Write-Host "🔨 Initializing git submodule..." -NoNewline
+    git submodule update --init >$null 2>&1
```
Comment (aslushnikov) [inline build.ps1:36]: FWIW, replacing this line with `git clone https://github.com/microsoft/playwright && git checkout $UPSTREAM_SHA` would be equal! However, both are fine.

**Review by aslushnikov — COMMENTED** [review]
lgtm

**Pull request merged**